modules/service/src/main/scala/ch/epfl/bluebrain/nexus/commons/service/retryer/TaskRetry.scala

```diff
@@ -0,0 +1,102 @@
+package ch.epfl.bluebrain.nexus.commons.service.retryer
+
+import ch.epfl.bluebrain.nexus.commons.types.RetriableErr
+import monix.eval.Task
+import monix.execution.CancelableFuture
+import monix.execution.Scheduler.Implicits.global
+
+import scala.concurrent.Future
+import scala.concurrent.duration.{FiniteDuration, _}
+import scala.util.Random
+
+trait TaskRetry {
```

Comment:
Could we consider designing this as a `RetryOps`?
```
object RetryOps {
  final implicit class Retryable[A](source: () => Future[A]) extends AnyVal {
    def retry(max: Int)(implicit strategy: RetryStrategy): CancelableFuture[A] =
      RetryOps.retry(source, max, strategy)
  }

  final def retry[A](source: () => Future[A], maxRetries: Int, strategy: RetryStrategy): Future[A] = {
    def retryTask(task: Task[A], retry: Int, currentDelay: FiniteDuration): Task[A] = ???
    val task = Task.deferFuture { source() }
    retryTask(task, maxRetries, strategy.init).runAsync
  }
}
```
... such that we can make use of it like so:
```
implicit val strategy = Backoff(...)
val f = () => index(value)
f.retry(10)
```